core/src/heaviest_subtree_fork_choice.rs

```diff
@@ -449,22 +453,27 @@ impl HeaviestSubtreeForkChoice {
                 .remove(&current_node)
                 .expect("Node must exist in tree");
 
-            removed.push(current_node.0);
-            to_visit.extend(current_fork_info.children.into_iter());
+            to_visit.extend(current_fork_info.children.iter());
+            tree.fork_infos.insert(current_node, current_fork_info);
         }
 
-        if let Some(parent) = parent {
-            // Remove link from parent
-            let parent_fork_info = self
-                .fork_infos
-                .get_mut(&parent)
-                .expect("Parent must exist in fork infos");
-            parent_fork_info.children.remove(slot_hash_key);
-        } else {
-            warn!("Dumped root of tree {:?}", slot_hash_key);
-        }
+        // Remove link from parent
+        let parent_fork_info = self
+            .fork_infos
+            .get_mut(&parent)
+            .expect("Parent must exist in fork infos");
+        parent_fork_info.children.remove(slot_hash_key);
+
+        // Reset fork info
+        let root = tree
+            .fork_infos
+            .get_mut(slot_hash_key)
+            .expect("Slot hash key is the root must exist");
+        root.stake_voted_subtree = stake_voted_subtree;
+        root.stake_voted_at = stake_voted_at;
+        root.latest_invalid_ancestor = None;
```

Comment:
This will be hard to maintain if we add more fields to `ForkInfo` in the future. I think instead we should build a new constructor  for `HeaviestSubtreeForkChoice`, can be called `new_from_split(fork_info: &ForkInfo) -> Self {
       ForkInfo {
           //explicitly list all fields so we don't forget to update them if we add new ones
      }
}`

Reply:
Not sure I follow exactly - HeaviestSubtreeForkChoice contains a hashmap of ForkInfo, but I rewrote it to make this more maintainable by keeping the original fork info, lmk

Reply:
If you add a field to `ForkInfo`, we will not necesarily remember that it may need to be updated here where all these other fields are being set.

Reply:
wdyt about how i did it here, where i no longer set fields.

Reply:
works for me!